File: Axiom/Internal/Ast/ConditionalExpression.cs
namespace Axiom.Internal.Ast
{
    internal class ConditionalExpression : Expression
    {
        public ConditionalExpression(Expression test, Expression consequent, Expression alternate)
        {
            Test = test;
            Consequent = consequent;
            Alternate = alternate;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public Expression Test { get; private set; }
        public Expression Consequent { get; private set; }
        public Expression Alternate { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/ConditionalExpression.java
package com.moybl.axiom.ast;

public class ConditionalExpression implements Expression {

	private Expression test;
	private Expression consequent;
	private Expression alternate;

	public ConditionalExpression(Expression test, Expression consequent, Expression alternate) {
		this.test = test;
		this.consequent = consequent;
		this.alternate = alternate;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public Expression getTest() {
		return test;
	}

	public Expression getConsequent() {
		return consequent;
	}

	public Expression getAlternate() {
		return alternate;
	}

}
